src/Polly.Core/Simmy/Outcomes/OutcomeChaosPipelineBuilderExtensions.TResult.cs

```diff
@@ -0,0 +1,157 @@
+        {
+            Enabled = enabled,
+            InjectionRate = injectionRate,
+            OutcomeGenerator = (_) => faultGenerator()
+        });
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds a fault chaos strategy to the builder.
+    /// </summary>
+    /// <typeparam name="TResult">The type of result the retry strategy handles.</typeparam>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="options">The fault strategy options.</param>
+    /// <returns>The builder instance with the retry strategy added.</returns>
+    public static ResiliencePipelineBuilder<TResult> AddChaosFault<TResult>(this ResiliencePipelineBuilder<TResult> builder, OutcomeStrategyOptions<Exception> options)
+    {
+        Guard.NotNull(builder);
+        Guard.NotNull(options);
+
+        builder.AddFaultCore<TResult>(options);
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds an outcome chaos strategy to the builder.
+    /// </summary>
+    /// <typeparam name="TResult">The type of result the retry strategy handles.</typeparam>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="enabled">A value that indicates whether or not the chaos strategy is enabled for a given execution.</param>
+    /// <param name="injectionRate">The injection rate for a given execution, which the value should be between [0, 1].</param>
+    /// <param name="result">The outcome to inject.</param>
+    /// <returns>The builder instance with the retry strategy added.</returns>
+    public static ResiliencePipelineBuilder<TResult> AddChaosResult<TResult>(this ResiliencePipelineBuilder<TResult> builder, bool enabled, double injectionRate, TResult result)
+    {
+        Guard.NotNull(builder);
+
+        builder.AddOutcomeCore<TResult, OutcomeStrategyOptions<TResult>>(new OutcomeStrategyOptions<TResult>
+        {
+            Enabled = enabled,
+            InjectionRate = injectionRate,
+            Outcome = new(result)
+        });
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds an outcome chaos strategy to the builder.
+    /// </summary>
+    /// <typeparam name="TResult">The type of result the retry strategy handles.</typeparam>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="enabled">A value that indicates whether or not the chaos strategy is enabled for a given execution.</param>
+    /// <param name="injectionRate">The injection rate for a given execution, which the value should be between [0, 1].</param>
+    /// <param name="outcomeGenerator">The outcome generator delegate.</param>
+    /// <returns>The builder instance with the retry strategy added.</returns>
+    public static ResiliencePipelineBuilder<TResult> AddChaosResult<TResult>(
+        this ResiliencePipelineBuilder<TResult> builder, bool enabled, double injectionRate, Func<ValueTask<Outcome<TResult>?>> outcomeGenerator)
```

Comment:
```suggestion
        this ResiliencePipelineBuilder<TResult> builder, bool enabled, double injectionRate, Func<TResult?>> outcomeGenerator)
```

Or just keep the above one.